src/library/playlistfeature.h

```diff
@@ -40,8 +40,9 @@ class PlaylistFeature : public BasePlaylistFeature {
     void slotPlaylistTableRenamed(int playlistId, QString a_strName);
 
  protected:
-    void buildPlaylistList();
-    void decorateChild(TreeItem *pChild, int playlist_id);
+    void buildPlaylistList() override;
+    void updatePlaylistList(int playlist_id) override;
```

Comment:
I don't understand what the difference between *buildPlaylistList* and *updatePlaylistList* is. Looking at their implementation both functions do fundamentally different things despite their very similar names.

Reply:
buildPlaylistList(), the original function clears the list of playlists m_playlistList and populates it again with fresh data from the database.  This function was also used to refresh single tree view items, without refreshing the whole tree view with the recently fetched database data. 
updatePlaylistList(int playlist_id) now only fetches the requested playlist and updates this in m_playlistList and GUI. 
This way both lists are still consistent and immune against external changes, the root cause of the crash. 

Should I rename updatePlaylistList(int playlist_id) to refreshPlaylistInPlaylistList((int playlist_id)? 
Or do you have a better idea to rename both?

Reply:
The whole naming is confusing, i.e. "Playlist" vs. "List". But leave it as it is to keep the impact low.

refresh/reloadPlaylistInPlaylistList works for me.

List -> Listing, Summary, Overview, ... someone else should wrap their head around this.